### PR TITLE
gameroyadvance

### PR DESCRIPTION
**Pull request by ROYMEETSWORLD**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keystonecaps/gameroyadvance/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Game Roy ADVANCE
+
+![Game Roy ADVANCE](https://user-images.githubusercontent.com/63612683/162093240-656e8740-f73c-454d-8b30-d60db9da212f.jpg)
+![GRALAYOUTS](https://user-images.githubusercontent.com/63612683/162095125-48746ddd-9b80-4f82-aea0-9a0d02c0e62e.JPG)
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/README.md:4]: Please rehost these images on Imgur, or simular image hosting sit.

File: keyboards/keystonecaps/gameroyadvance/config.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/config.h:68]:
```suggestion
```

File: keyboards/keystonecaps/gameroyadvance/config.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/config.h:57]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/keystonecaps/gameroyadvance/config.h [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
+
+/* Change USB Polling Rate to 1000hz and a larger keys per scan for elite gaming*/
+#define USB_POLLING_INTERVAL_MS 1
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/config.h:74]: Is the default. 
```suggestion
```

File: keyboards/keystonecaps/gameroyadvance/config.h [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/config.h:71]: Deprecated a LONG time ago.
```suggestion
```

File: keyboards/keystonecaps/gameroyadvance/config.h [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+/* Change USB Polling Rate to 1000hz and a larger keys per scan for elite gaming*/
+#define USB_POLLING_INTERVAL_MS 1
+#define QMK_KEYS_PER_SCAN 12
+
+#define RGB_DI_PIN C7
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/config.h:79]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/keystonecaps/gameroyadvance/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    }
+    return false;
+}
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/keymaps/default/keymap.c:55]:
```suggestion
```

File: keyboards/keystonecaps/gameroyadvance/gameroyadvance.c [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "gameroyadvance.h"
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/gameroyadvance.c:18]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "gameroyadvance.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

File: keyboards/keystonecaps/gameroyadvance/keymaps/default_all/keymap.c [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    }
+    return false;
+}
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/keymaps/default_all/keymap.c:55]:
```suggestion
```

File: keyboards/keystonecaps/gameroyadvance/rules.mk [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+ENCODER_ENABLE = yes
+SPLIT_KEYBOARD = yes
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/rules.mk:22]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output

ENCODER_ENABLE = yes
SPLIT_KEYBOARD = yes
```

File: keyboards/keystonecaps/gameroyadvance/README.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
+# Game Roy ADVANCE
```
Comment (drashna) [inline keyboards/keystonecaps/gameroyadvance/README.md:1]: Pleaes rename this file to be lower case.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by ROYMEETSWORLD** [comment]
I believe I made all the changes that you suggested. Thank you again for your time.

**Review by drashna — APPROVED** [review]

File: keyboards/keystonecaps/gameroyadvance/readme.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
+# Game Roy ADVANCE
+
+![Game Roy ADVANCE](https://i.imgur.com/N6RI4fU.jpg)
```
Comment (fauxpark) [inline keyboards/keystonecaps/gameroyadvance/readme.md:3]:
```suggestion
![Game Roy ADVANCE](https://i.imgur.com/N6RI4fUh.jpg)
```

File: keyboards/keystonecaps/gameroyadvance/readme.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+```
+make keystonecaps/gameroyadvance:default
+```
+
+Flashing example for this keyboard:
+
+```
+make keystonecaps/gameroyadvance:default:flash
+```
```
Comment (fauxpark) [inline keyboards/keystonecaps/gameroyadvance/readme.md:22]:
```suggestion
    make keystonecaps/gameroyadvance:default

Flashing example for this keyboard:

    make keystonecaps/gameroyadvance:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by ROYMEETSWORLD** [comment]
@fauxpark is there something else you would like me to change?

**Review by fauxpark — APPROVED** [review]

**Comment by ROYMEETSWORLD** [comment]
Thank you for your time. I'll make those changes as soon as I can.

**Pull request merged**